### PR TITLE
Implement client side form validation using javascript

### PR DESCRIPTION
**Pull request by zainsadaqat**
### In this pull request, I implemented the following features:

- When the user submits the form, you check if the email is in lower case.
- If the validation is OK, the form is sent.
- If the validation is not OK, you show an error message to the user near the submit button informing them of the error and the form is not sent.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -15,6 +15,34 @@ mobileMenuItems.addEventListener('click', () => {
...
+    li.style.color = 'red';
+    li.style.marginBottom = '16px';
+    li.style.padding = '14px';
+    errorMessagesContainer.append(li);
+  });
+  errorMessagesContainer.classList.toggle('d-none');
+};
+
+form.addEventListener('submit', (e) => {
+  const errorMessages = [];
```
Comment (Shaher-11) [inline script.js:36]: Kindly implement an error message for  entering uppercase letters in the mail field

**Review by Shaher-11 — CHANGES_REQUESTED** [review]
#### Hello @zainsadaqat,

### Great job so far!
You have been doing an incredible  job on the project so far:clap::clap:
The project is very well structured:smiley::smiley:
However, we still need to make a few changes to get the project approved:muscle:

 ## Required Changes ♻️

- [x] Check the comment below:point_down:


>As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Happy Coding!👏👏👏

File: script.js (hunk 1 above)
Comment (zainsadaqat) [reply script.js:36]: Can you please elaborate a little bit more?

Comment (Shaher-11) [reply script.js:36]: when i fill the email field with capital letters the form does not give the error message as it should as the email field should accept only small letters

Comment (zainsadaqat) [reply script.js:36]: https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/9_validate_form.md

In the above link, they asked us to show the error on submit that's why It'll show the error on submit and not before submit.

Comment (Shaher-11) [reply script.js:36]: It does not show an error at all

Comment (zainsadaqat) [reply script.js:36]: ## Kindly, check below the button

Screenshot from 2021-07-31 02-13-47

Comment (Shaher-11) [reply script.js:36]: submit the PR and it will approved

Comment (zainsadaqat) [reply script.js:36]: I've to submit for a second review? But it's still showing changes required.

Comment (Shaher-11) [reply script.js:36]: You need to submit the code review request so i can approve it

Comment (zainsadaqat) [reply script.js:36]: But I already did, Kindly tell me If you're asking for a second review request?

Comment (Shaher-11) [reply script.js:36]: Yes please

Comment (zainsadaqat) [reply script.js:36]: Alright, I submitted the PR.

Comment (rloterh) [reply script.js:36]: Issue not resolved. The email field  of the form is accepting uppercase letters without displaying error message  Kindly take some time to implement this changes requested by the previous reviewer.

Comment (zainsadaqat) [reply script.js:36]: Kindly, explain where it's not working. I checked it several times applying different values both upper and lower. and It doesn't let pass the upper case value.

Comment (zainsadaqat) [reply script.js:36]: ### Kindly, check below the button. They asked us to create error message near to submit button and this is what I did.
Screenshot from 2021-07-31 02-13-47

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Zain,

You did great on this project, and for making the changes requested by the previous reviewer!👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there!👍 

## Good Points 👍  
- No linter errors

## Required Changes ♻️
- [x] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [OPTIONAL]
You can improve the `Demo Link` section of your ReadMe by concealing the actual url and providing just a clickable link, by doing this in your ReadMe markdown code: `[Demo Link:](https://danniemo.github.io/your portfolio)`

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashbaord. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Comment by zainsadaqat** [comment]
> Hi Zain,
> 
> You did great on this project, and for making the changes requested by the previous reviewer! 
> There are however a few issues that you still need to work on to go to the next project but you are almost there!
> 
> ## Good Points 
> * No linter errors
> 
> ## Required Changes 
> * [ ]  _Check the comments under the review._
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
> You can also consider:
> 
> * [OPTIONAL]
>   You can improve the `Demo Link` section of your ReadMe by concealing the actual url and providing just a clickable link, by doing this in your ReadMe markdown code: `[Demo Link:](https://danniemo.github.io/your portfolio)`
> 
> Cheers and Happy coding!  
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by zainsadaqat** [comment]
# Thank @arslanbisharat for approving this pull request!

**Pull request merged**